sdk/confidentialledger/Azure.Storage.ConfidentialLedger/README.md

```diff
@@ -42,69 +40,241 @@ export AZURE_TENANT_ID="tenant id"
 
-TODO
+```C# Snippet:GetReceipt
+Response receiptResponse = ledgerClient.GetReceipt(transactionId);
+string receiptJson = new StreamReader(receiptResponse.ContentStream).ReadToEnd();
+
+Console.WriteLine(receiptJson);
+```
 
 #### Sub-ledgers
 While most use cases will involve one ledger, we provide the sub-ledger feature in case different logical groups of data need to be stored in the same Confidential Ledger.
 
-TODO
+```C# Snippet:SubLedger
+ledgerClient.PostLedgerEntry(
+    RequestContent.Create(
+        new { contents = "Hello from Chris!", subLedgerId = "Chris' messages" }));
+
+ledgerClient.PostLedgerEntry(
+    RequestContent.Create(
+        new { contents = "Hello from Allison!", subLedgerId = "Allison's messages" }));
+```
 
 When no sub-ledger id is specified on method calls, the Confidential Ledger service will assume a constant, service-determined sub-ledger id.
 
-TODO
+```C# Snippet:NoSubLedgerId
+Response postResponse = ledgerClient.PostLedgerEntry(
+    RequestContent.Create(
+        new { contents = "Hello world!" }));
+postResponse.Headers.TryGetValue("x-ms-ccf-transaction-id", out string transactionId);
+string subLedgerId = JsonDocument.Parse(statusResponse.Content)
+    .RootElement
+    .GetProperty("subLedgerId")
+    .GetString();
+
+// Provide both the transactionId and subLedgerId.
+Response getBySubledgerResponse = ledgerClient.GetLedgerEntry(transactionId, subLedgerId);
+
+string contents = JsonDocument.Parse(getBySubledgerResponse.Content)
+    .RootElement
+    .GetProperty("contents")
+    .GetString();
+
+Console.WriteLine(contents); // "Hello world!"
+
+// Now just provide the transactionId.
+getBySubledgerResponse = ledgerClient.GetLedgerEntry(transactionId);
+
+string subLedgerId2 = JsonDocument.Parse(getBySubledgerResponse.Content)
+    .RootElement
+    .GetProperty("subLedgerId")
+    .GetString();
+
+Console.WriteLine($"{subLedgerId} == {subLedgerId2}");
+```
 
 Ledger entries are retrieved from sub-ledgers. When a transaction id is specified, the returned value is the value contained in the specified sub-ledger at the point in time identified by the transaction id. If no transaction id is specified, the latest available value is returned.
 
-TODO
+```C# Snippet:GetEnteryWithNoTransactionId
+Response firstPostResponse = ledgerClient.PostLedgerEntry(
+    RequestContent.Create(new { contents = "Hello world 0" }));
+ledgerClient.PostLedgerEntry(
+    RequestContent.Create(new { contents = "Hello world 1" }));
+Response subLedgerPostResponse = ledgerClient.PostLedgerEntry(
+    RequestContent.Create(new { contents = "Hello world sub-ledger 0" }),
+    "my sub-ledger");
+ledgerClient.PostLedgerEntry(
+    RequestContent.Create(new { contents = "Hello world sub-ledger 1" }),
+    "my sub-ledger");
+
+firstPostResponse.Headers.TryGetValue("x-ms-ccf-transaction-id", out string transactionId);
```

Comment:
`x-ms-ccf-transaction-id` gets used enough in the examples that I wonder if we should consider exposing it via some kind of `ConfidentialLedgerHeaders` constants class.